packages/grid_client/src/clients/kyc/client.ts

```diff
@@ -0,0 +1,190 @@
+ * It allows fetching verification data, status, and token by preparing necessary headers
+ * and sending requests to the specified API domain.
+ *
+ * @class KYC
+ * @example
+ * ```typescript
+ * const kyc = new KYC("https://api.example.com", KeypairType.sr25519, "mnemonic");
+ * const data = await kyc.data();
+ * const status = await kyc.status();
+ * const token = await kyc.getToken();
+ * ```
+ * @param {string} apiDomain - The API domain for the KYC service.
+ * @param {KeypairType} [keypairType=KeypairType.sr25519] - The type of keypair to use.
+ * @param {string} mnemonic - The mnemonic for generating the keypair.
+ * @method data - Fetches the verification data from the KYC service.
+ * @method status - Fetches the verification status from the KYC service.
+ * @method getToken - Fetches the token from the KYC service.
+ */
+export class KYC {
+  private keyring: Keyring;
+  private keypair: KeyringPair;
+  public address: string;
+  /**
+   * Creates an instance of KYC.
+   * @param apiDomain - The API domain for the TFGrid KYC service.
+   * @param keypairType - The type of keypair to use (default is sr25519).
+   * @param mnemonic - The mnemonic for generating the keypair.
+   */
+  constructor(
+    public apiDomain: string,
+    private mnemonic: string,
+    public keypairType: KeypairType = KeypairType.sr25519,
+  ) {
+    if (mnemonic === "") {
+      throw new ValidationError("mnemonic is required");
+    }
+    /** The api domain should not contains any prefix or postfix */
+    this.apiDomain = apiDomain.replace("https://", "");
+    this.apiDomain = apiDomain.replace("http://", "");
+    this.apiDomain = apiDomain.replace("/", "");
+  }
+
+  /**
+   * Setup the keypair and the address
+   *
+   * @returns {Promise<void>}
+   * @private
+   */
+  private async setupKeyring() {
+    const keyring = new Keyring({ type: this.keypairType });
+    await waitReady();
+    this.keypair = keyring.addFromUri(this.mnemonic);
+    this.address = this.keypair.address;
+  }
+
+  /**
+   * Prepares the headers required for TFGrid KYC requests.
+   *
+   * This method generates a set of headers that include a timestamp-based challenge,
+   * a signed challenge using the user's key, and other necessary information.
+   *
+   * @returns {Promise<Record<string, string>>} A promise that resolves to an object containing the prepared headers.
+   *
+   */
+  private async prepareHeaders(): Promise<Record<string, string>> {
+    await this.setupKeyring();
+    const timestamp = Date.now();
+    const challenge = stringToHex(`${this.apiDomain}:${timestamp}`);
+    const signedChallenge = this.keypair.sign(bytesFromHex(challenge));
+    const signedMsgHex = Buffer.from(signedChallenge).toString("hex");
+
+    const headers: KycHeaders = {
+      "content-type": "application/json",
+      "X-Client-ID": this.address,
+      "X-Challenge": challenge,
+      "X-Signature": signedMsgHex,
+    };
+    return headers as unknown as Record<string, string>;
```

Comment:
unknown?

Reply:
Yes we have to convert the type to record so we have to use it [ref](https://stackoverflow.com/questions/69399211/typescript-why-does-as-unknown-as-x-work)
What do you suggest?